src/dsp_tools/xmllib/models/geometry.py

```diff
@@ -0,0 +1,102 @@
+        if not is_decimal(line_width):
+            problems.append(line_msg)
+        elif not 0 < int(line_width) <= 5:
+            problems.append(line_msg)
+        if not is_color(color):
+            problems.append(f"The color must be a valid hexadecimal color. The provided value is not valid: {color}")
+        if status.lower() not in ["active", "deleted"]:
+            problems.append(
+                f'The status must be either "active" or "deleted". The provided value is not valid: {status}'
+            )
+        if type_.lower() not in ["circle", "rectangle", "polygon"]:
+            problems.append(
+                f'The type must be either ""circle", "rectangle" or "polygon" . '
+                f"The provided value is not valid: {type_}"
+            )
+        if problems:
+            warnings.warn(
+                DspToolsUserWarning(
+                    f"Some of the shape values of the resource with the ID '{self.resource_id}' are not valid: "
+                    f"\n- {'\n- '.join(problems)}"
+                )
+            )
+        self.line_width = line_width
+        self.color = color
+        self.status = status.lower()
+        self.type_ = type_.lower()
+        return self
+
+    def to_json_string(self) -> str:
+        if len(self.points) == 0:
+            msg = (
+                f"The region shape of the resource with the ID '{self.resource_id}' does not have any points. "
+                f"At least two points are required."
+            )
+            warnings.warn(DspToolsUserWarning(msg))
+        sorted_points = sorted(self.points, key=lambda x: x.x)
+        points = [x.to_dict() for x in sorted_points]
+        json_dict = {
+            "status": self.status,
+            "type": self.type_,
+            "lineWidth": self.line_width,
+            "points": points,
+        }
+        return json.dumps(json_dict)
+
+
+@dataclass
+class GeometryPoint:
+    x: float
+    y: float
+    resource_id: str
+
+    def __post_init__(self) -> None:
+        if not all([is_decimal(self.x), is_decimal(self.y)]):
+            msg = (
+                f"The entered geometry points for the resource with the ID '{self.resource_id}' are not floats. "
+                f"x: '{self.x}', y: '{self.y}'"
+            )
+            warnings.warn(DspToolsUserWarning(msg))
+        else:
+            info = []
+            if not 0 <= float(self.x) <= 1:
+                info.append(f"x: '{self.x}'")
+            if not 0 <= float(self.y) <= 1:
+                info.append(f"y: '{self.y}'")
+            if info:
+                msg = (
+                    f"The geometry points must be larger/equal 0 and smaller/equal 1. "
+                    f"The following points of the resource with the ID '{self.resource_id}' "
+                    f"are not valid: {', '.join(info)}"
+                )
+                warnings.warn(DspToolsUserWarning(msg))
+            self.x = float(self.x)
```

Comment:
I am "converting it" in case someone entered a float as a string. Because in the json we do not want them with quotation marks.